src/Raven.Server/Documents/Indexes/Static/Spatial/SpatialField.cs

```diff
@@ -71,21 +81,41 @@ private static SpatialStrategy CreateStrategy(string fieldName, SpatialOptions o
             return null;
         }
 
-        public AbstractField[] CreateIndexableFields(object value)
+        public AbstractField[] LuceneCreateIndexableFields(object value)
         {
             var shape = value as IShape;
             if (shape != null || TryReadShape(value, out shape))
             {
                 var fields = Strategy.CreateIndexableFields(shape);
                 Array.Resize(ref fields, fields.Length + 1);
-                fields[fields.Length - 1] = new Field(Constants.Documents.Indexing.Fields.SpatialShapeFieldName, WriteShape(shape), Field.Store.YES, Field.Index.NO);
+                fields[^1] = new Field(Constants.Documents.Indexing.Fields.SpatialShapeFieldName, WriteShape(shape), Field.Store.YES, Field.Index.NO);
 
                 return fields;
             }
 
             return Array.Empty<AbstractField>();
         }
 
+        public CoraxSpatialPointEntry[] CoraxCreateIndexableFields(object value)
+        {
+            var shape = value as IShape;
+            if (shape != null || TryReadShape(value, out shape))
+            {
+                if (shape.HasArea)
+                    throw new NotSupportedException("We do not support indexing of figure.");
```

Comment:
introduce NotSupportedInCorax exception (similar to https://github.com/ravendb/ravendb/blob/v6.0/src/Raven.Client/Exceptions/Sharding/NotSupportedInShardingException.cs), add a better message here.

Reply:
done